### PR TITLE
Fix CI failing - invalid caches

### PR DESCRIPTION
**Pull request by b-onc**
In the hopes that this fixes the CI
Let's see if the CI passes!

### CODE REVIEW

File: carthage.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# the build will fail on lipo due to duplicate architectures.
+
+CURRENT_XCODE_VERSION=$(xcodebuild -version | grep "Build version" | cut -d' ' -f3)
+echo "EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_$CURRENT_XCODE_VERSION = arm64 arm64e armv7 armv7s armv6 armv8" >> $xcconfig
+
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200 = $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_$(XCODE_PRODUCT_BUILD_VERSION))' >> $xcconfig
+echo 'EXCLUDED_ARCHS = $(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))' >> $xcconfig
+
+export XCODE_XCCONFIG_FILE="$xcconfig"
+carthage "$@"
```
Comment (VojtaStavik) [inline carthage.sh:19]: If I'm not mistaken we already have this in the repo as `sample-bootstrap.sh`

Comment (VojtaStavik) [reply carthage.sh:19]: @b-onc feel free to rename it to something more general if you want btw

File: carthage.sh [hunk 2]
```diff
@@ -1,4 +1,5 @@
-#!/usr/bin/env bash
+# carthage.sh
+# Usage example: ./carthage.sh build --platform iOS
```
Comment (VojtaStavik) [inline carthage.sh:2]: Please also update https://github.com/GetStream/stream-chat-swift/blob/main/Sample_v3/README.md with the new script name

Comment (b-onc) [reply carthage.sh:2]: Updated. We don't actually use carthage for v3 anymore

Comment (VojtaStavik) [reply carthage.sh:2]: Good, we can probably remove the whole section from the readme then.

Comment (b-onc) [reply carthage.sh:2]: I did, please check

Comment (VojtaStavik) [reply carthage.sh:2]: Good. Now after you actually pushed the changes, it's easier for me to review them :trollface:

**Review by VojtaStavik — APPROVED** [review]

**Pull request merged**